### PR TITLE
Friend add rank and stock holdings to friend search

### PR DESCRIPTION
**Pull request by kingsleyallen**
- Added the rank and stock selections for each user in the search
- Restyled accordingly (and added hover effect)
- Search field now ignores any spaces on edges or within username

Screen Shot 2021-09-24 at 6 40 41 PM

### CODE REVIEW

File: client/dist/styles.css [hunk 1]
```diff
@@ -55,7 +55,12 @@ a, a:hover, a:focus, a:active {
 /**********END BASE**********/
 /**********BEGIN LAYOUT**********/
 
+body {
```
Comment (kingsleyallen) [inline client/dist/styles.css:58]: To prevent app from breaking horribly if made very narrow

File: db/setup.sql [hunk 2]
```diff
@@ -83,7 +83,8 @@ VALUES
 ('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A22', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A12', 'amzn', 'nasdaq', 100000000),
 ('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A23', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A13', 'fb', 'nasdaq', 100000000),
 ('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A24', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A14', 'msft', 'nasdaq', 100000000),
-('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A25', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A15', 'brk.a', 'nyse', 100000000);
+('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A25', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A15', 'brk.a', 'nyse', 100000000),
+('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A26', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A15', 'nflx', 'nasdaq', 500);
```
Comment (kingsleyallen) [inline db/setup.sql:87]: Added a second holding to user "gramps" in db setup so you can see multiple stock holdings right away in the friend search

File: db/setup.sql [hunk 3]
```diff
@@ -112,4 +113,14 @@ VALUES
 ('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A12', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A13'),
 ('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A13', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A14'),
 ('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A14', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A15'),
-('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A15', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11');
\ No newline at end of file
+('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A15', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11');
+
```
Comment (kingsleyallen) [inline db/setup.sql:117]: Created a db view of users and their rank and then joined to it in the db qeury

File: client/src/components/Friend.jsx [hunk 4]
```diff
@@ -23,15 +23,23 @@ const Friend = () => {
   }, [param]);
 
   const handleUserInput = (e) => {
-    setParam(!e.target.value ? null : e.target.value);
+    const searchVal = e.target.value.replace(/\s/g, '');
```
Comment (baudot) [inline client/src/components/Friend.jsx:26]: Nice use of a regex.

File: db/setup.sql (hunk 2 above)
Comment (baudot) [reply db/setup.sql:87]: 👍

File: server/db/queries.js [hunk 5]
```diff
@@ -28,8 +28,19 @@ class Db {
...
+        ranked_users.user_rank,
+        string_agg(UPPER(positions.ticker_symbol), ', ') as holdings
+      FROM users
+      INNER JOIN ranked_users ON
+        users.id = ranked_users.user_id
+      LEFT JOIN positions ON
+        users.id = positions.user_id
+      WHERE LOWER(users.username) LIKE LOWER('%${usernameFragment}%')
+      GROUP BY users.id, users.username, ranked_users.user_rank
+      ORDER BY users.username ASC;
```
Comment (baudot) [inline server/db/queries.js:43]: Well this query grew up.

**Review by baudot — APPROVED** [review]

**Pull request merged**